js/pricing.js

```diff
@@ -101,27 +101,39 @@ $(
         /**
          * Submits consent transaction.
          *
-         * <p>Prepares Privacy, Support Agreement consents data and transaction Url for sending and
-         * calls send transaction function.
+         * <p>Prepares transaction Url, Privacy and Support Agreement consents data, generates
+         * the reCaptcha token and adds it to the request data. After that calls send transaction
+         * function.
          */
         function submitOrder() {
+            const transactionUrl = `${apiUrl}/transaction`;
             const privacyConsent = $privacyConsent.prop("checked");
             const supportAgreementConsent = $supportAgreementConsent.prop("checked");
-            const data = JSON.stringify({privacyConsent, supportAgreementConsent});
-            const transactionUrl = `${apiUrl}/transaction`;
-            sendPaymentTransaction(transactionUrl, data);
+
+            showRedirect();
+
+            grecaptcha.ready(function() {
+                grecaptcha.execute("{{site.data.payment_config.reCaptchaSiteKey}}", {action: 'submitPaymentTransaction'}).then(function(token) {
+                    let data = {privacyConsent, supportAgreementConsent};
+                    data.token = token;
+                    data = JSON.stringify(data);
+                    sendPaymentTransaction(transactionUrl, data);
+                });
+            });
         }
 
         /**
-         * The user's consent.
+         * The user's consent and reCaptcha token.
```

Comment:
I'd better have a new Request object with the reCAPTCHA token as one field and the consent as the other.